Src/FluentAssertions/Specialized/AsyncFunctionAssertions.cs

```diff
@@ -104,22 +101,30 @@ public void NotThrow<TException>(string because = "", params object[] becauseArg
             }
             catch (Exception exception)
             {
-                while (exception is AggregateException)
-                {
-                    exception = exception.InnerException;
-                }
+                var deepestException = GetDeepestException(exception);
```

Comment:
🙃 The conventions avoid the use of `var`, unless the type is explicitly stated as a generic parameter.